lib/output.js

```diff
@@ -117,15 +117,37 @@ function OutputStream(options) {
         function quote_double() {
             return '"' + str.replace(/\x22/g, '\\"') + '"';
         }
+        function quote_template() {
+            if (!options.ascii_only) {
+                str = str.replace(/\\(r|n|u2028|u2029)/g, function(s, c) {
+                    switch(c) {
+                        case "n": return "\n";
+                        case "r": return "\r";
+                        case "u2028": return "\u2028";
+                        case "u2029": return "\u2029";
+                    }
+                    return "\\" + c;
+                });
+            }
+            return '`' + str.replace(/`/g, '\\`') + '`';
+        }
         if (options.ascii_only) str = to_ascii(str);
         switch (options.quote_style) {
           case 1:
+            if (quote == "`" && sq > 0)
+                return quote_template();
             return quote_single();
           case 2:
+            if (quote == "`" && dq > 0)
+                return quote_template();
             return quote_double();
           case 3:
+            if (quote == "`")
+                return quote_template();
             return quote == "'" ? quote_single() : quote_double();
           default:
+            if (quote == "`" && dq > 0 && sq > 0)
```

Comment:
Maybe better to not concentrate on that bit of eye candy and get rid of it now. I've got that part implemented before the compressor actually worked and make me realise the codegen isn't a compressor after all :/.